packages/wallets/src/components/Base/WalletButton/WalletButton.tsx

```diff
@@ -1,35 +1,81 @@
-import React, { ComponentProps, CSSProperties } from 'react';
+import React, { ButtonHTMLAttributes } from 'react';
 import classNames from 'classnames';
+import { TGenericSizes } from '../types';
+import { WalletText } from '../WalletText';
 import styles from './WalletButton.module.css';
 
-interface WalletButtonProps extends ComponentProps<'button'> {
-    children: React.ReactNode;
-    color?: CSSProperties['color'] | 'primary-light' | 'primary';
+interface WalletButtonProps extends ButtonHTMLAttributes<HTMLButtonElement> {
+    color?: 'black' | 'primary-light' | 'primary' | 'white';
+    icon?: React.ReactNode;
```

Comment:
```suggestion
    icon?: ReactElement;
```
Since ideally icon should only be an element